components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -0,0 +1,251 @@
+
+    /**
+     * Delete the resource for the given resourceId
+     *
+     * @param resourceId resourceId of the resource which need to get deleted
+     * @return Response with the status of resource deletion
+     */
+    @Override
+    public Response deleteResource(String resourceId) {
+
+        try {
+            if (!isResourceId(resourceId)) {
+
+                try {
+                    if (ResourceUtils.getResourceService().deleteResource(resourceId)) {
+                        return Response.status(Response.Status.NO_CONTENT).build();
+                    }
+
+                } catch (UMAServiceException e) {
+                    handleErrorResponse(e, log);
+                    log.error("Invalid request. ", e);
+                } catch (Throwable throwable) {
+                    log.error("Internal server error occurred. ", throwable);
+                }
+            }
+        } catch (UMAClientException e) {
+            handleErrorResponse(e, log);
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when retrieving resource from the resource server.", e);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Validate resourceId
+     *
+     * @param umaException exception that should pass to handle error.
+     */
+    public static void handleErrorResponse(UMAException umaException, Log log) throws ResourceEndpointException {
+
+        ResourceEndpointConstants resourceEndpointConstants = new ResourceEndpointConstants();
+        if (resourceEndpointConstants.getResponseMap().containsKey(umaException.getErrorCode())) {
+            String statusCode = resourceEndpointConstants.getResponseMap().get(umaException.getErrorCode())[0];
+            Response.Status status = Response.Status.fromStatusCode(Integer.parseInt(statusCode));
+            String errorCode = resourceEndpointConstants.getResponseMap().get(umaException.getErrorCode())[1];
+
+            throw buildResourceEndpointException(status, errorCode, umaException.getMessage());
+        }
+    }
+
+    private static ResourceEndpointException buildResourceEndpointException(Response.Status status,
+                                                                            String code, String description) {
+
+        ErrorDTO errorDTO = new ErrorDTO();
+        errorDTO.setCode(code);
+        errorDTO.setDescription(description);
+        return new ResourceEndpointException(status, errorDTO);
+    }
+
+    /**
+     * Validate resourceId
+     *
+     * @param resourceId resourceId of the resource which need to get deleted
+     * @return validator obtain
+     */
+    public boolean isResourceId(String resourceId) throws UMAClientException {
+
+        boolean validator = true;
+        String pattern = "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$";
+        Pattern pat = Pattern.compile(pattern);
+        Matcher match = pat.matcher(resourceId);
+
+        if (match.find()) {
```

Comment:
You can directly return the result for match.find()

Reply:
Done.